### PR TITLE
Update Globalnet doc to include hostnetwork support

### PR DESCRIPTION
**Pull request by sridhargaddam**
This commit also clarifies that only POD to remote service
via globalIp is supported by Globalnet.

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

File: src/content/architecture/globalnet/_index.en.md [hunk 1]
```diff
@@ -23,30 +23,38 @@ This is largely problematic because most actual deployments use the default CIDR
...
 
-Once configured, each service and pod that requires cross-cluster access is allocated an IP, named `GlobalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This `GlobalIp` is used for all cross-cluster communication to and from to this Pod/Service. Routing and IPTable/OVS/OVN rules are configured to use this IP for ingress and egress. All address translations occur on the Gateway node.
+Once configured, each service and pod that requires cross-cluster access is allocated an IP, named `globalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This `globalIp` is used for all cross-cluster communication to and from to the Pod to globalIp of remote Service. Routing and IPTable/OVS/OVN rules are configured to use this IP for ingress and egress. All address translations occur on the Gateway node.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:26]:
```suggestion
Once configured, each Service and Pod that requires cross-cluster access is allocated an IP, named `globalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This globalIp is used for all cross-cluster communication to and from a Pod and the globalIp of a remote Service. Routing and IPTable/OVS/OVN rules are configured to use the globalIp for ingress and egress. All address translations occur on the Gateway node.
```

I think we need to be consistent with our use of case with pods/services. There's a mix of _pod_, _Pod_, and _POD_. We need to pick one. Looking at the k8s docs, they consistently use _Pod_ and _Service_.

File: src/content/architecture/globalnet/_index.en.md [hunk 2]
```diff
@@ -23,30 +23,38 @@ This is largely problematic because most actual deployments use the default CIDR
...
 
 ![Figure 1 - Proposed solution](/images/globalnet/overlappingcidr-solution.png)
 
+{{% notice info %}}
+
+Unlike vanilla Submariner, where POD to POD connectivity is also supported, Globalnet only supports POD to remote service connectivity using globalIps.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:32]:
```suggestion
Unlike vanilla Submariner, where Pod to Pod connectivity is also supported, Globalnet only supports Pod to remote Service connectivity using globalIps.
```

File: src/content/architecture/globalnet/_index.en.md [hunk 3]
```diff
@@ -23,30 +23,38 @@ This is largely problematic because most actual deployments use the default CIDR
...
+
+{{% /notice %}}
+
 ### submariner-globalnet
 
-Submariner GlobalNet is a component that provides cross-cluster connectivity between pods and services using their GlobalIps. Compiled as binary `submariner-globalnet`, it is responsible for maintaining a pool of global IPs, allocating IPs from the globalIp pool to pods and services, annotating services and pods with their GlobalIp, and configuring the required rules on the gateway node to provide cross-cluster connectivity using GlobalIps. It mainly consists of two key components: the IP Address Manager and Globalnet.
+Submariner GlobalNet is a component that provides cross-cluster connectivity from PODs to remote Services using their globalIps. Compiled as binary `submariner-globalnet`, it is responsible for maintaining a pool of global IPs, allocating IPs from the GlobalIp pool to pods and services, annotating services and pods with their globalIp, and configuring the required rules on the gateway node to provide cross-cluster connectivity using globalIps.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:38]:
```suggestion
Submariner GlobalNet is a component that provides cross-cluster connectivity from Pods to remote Services using their globalIps. Compiled as binary `submariner-globalnet`, it is responsible for maintaining a pool of global IPs, allocating IPs from the GlobalIp pool to pods and services, annotating Services and Pods with their globalIp, and configuring the required rules on the gateway node to provide cross-cluster connectivity using globalIps.
```

File: src/content/architecture/globalnet/_index.en.md [hunk 4]
```diff
@@ -23,30 +23,38 @@ This is largely problematic because most actual deployments use the default CIDR
...
+Globalnet also supports connectivity from the Nodes (including PODs that use HostNetworking) to globalIp of remoteServices.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:39]:
```suggestion
Globalnet also supports connectivity from the Nodes (including Pods that use HostNetworking) to globalIp of remote Services.
```

File: src/content/architecture/globalnet/_index.en.md [hunk 5]
```diff
@@ -23,30 +23,38 @@ This is largely problematic because most actual deployments use the default CIDR
...
 
 #### Globalnet
 
-This component is responsible for programming the routing entries and IPTable rules and does the following:
+This component is responsible for programming the routing entries, IPTable rules and does the following:
 
 * Creates initial IPTables chains for Globalnet rules.
-* Whenever a pod is annotated with Global IP, creates an egress SNAT rule to convert source ip from PodIp to pod's GlobalIp on the Gateway Node.
-* Whenever a service is annotated with Global IP, creates an ingress rule to direct all traffic destined to service's GlobalIp to the service's `kube-proxy` IPTables chain which in turn directs traffic to service's backend pods.
+* Whenever a pod is annotated with Global IP, creates an egress SNAT rule to convert source ip from PodIp to pod's globalIp on the Gateway Node.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:56]:
```suggestion
* Whenever a Pod is annotated with a globalIp, creates an egress SNAT rule to convert the source Ip from the Pod's Ip to the Pod's globalIp on the Gateway Node.
```

File: src/content/architecture/globalnet/_index.en.md [hunk 6]
```diff
@@ -23,30 +23,38 @@ This is largely problematic because most actual deployments use the default CIDR
...
+* Whenever a service is annotated with Global IP, creates an ingress rule to direct all traffic destined to service's globalIp to the service's `kube-proxy` IPTables chain which in turn directs traffic to service's backend pods.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:57]:
```suggestion
* Whenever a Service is annotated with a globalIp, creates an ingress rule to direct all traffic destined to the Service's globalIp to the Service's `kube-proxy` IPTables chain which in turn directs traffic to Service's backend Pods.
```

File: src/content/architecture/globalnet/_index.en.md [hunk 7]
```diff
@@ -55,7 +63,7 @@ Globalnet currently relies on `kube-proxy` and thus will only work with deployme
 
 Connectivity is only part of the solution as pods still need to know the IPs of services on remote clusters.
 
-This is achieved by enhancing [lighthouse](https://github.com/submariner-io/lighthouse) with support for GlobalNet. The Lighthouse controller adds the service's GlobalIp to the `MultiClusterService` object that is distributed to all clusters. The [lighthouse plugin](https://github.com/submariner-io/lighthouse/tree/master/plugin/lighthouse) then uses the service's GlobalIp when replying to DNS queries for the service.
+This is achieved by enhancing [lighthouse](https://github.com/submariner-io/lighthouse) with support for GlobalNet. The Lighthouse controller adds the service's globalIp to the `MultiClusterService` object that is distributed to all clusters. The [lighthouse plugin](https://github.com/submariner-io/lighthouse/tree/master/plugin/lighthouse) then uses the service's globalIp when replying to DNS queries for the service.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:66]:
```suggestion
This is achieved by enhancing [lighthouse](https://github.com/submariner-io/lighthouse) with support for GlobalNet. The Lighthouse controller adds the service's globalIp to the `MultiClusterService` object that is distributed to all clusters. The [lighthouse plugin](https://github.com/submariner-io/lighthouse/tree/master/plugin/lighthouse) then uses the Service's globalIp when replying to DNS queries for the Service.
```

File: src/content/architecture/globalnet/_index.en.md (hunk 1 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md]: Yeah makes sense @tpantelis.

File: src/content/architecture/globalnet/_index.en.md (hunk 2 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 3 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 4 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 5 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 6 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 7 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md]: Done

File: src/content/architecture/globalnet/_index.en.md [hunk 8]
```diff
@@ -23,30 +23,38 @@ This is largely problematic because most actual deployments use the default CIDR
...
 
-Once configured, each service and pod that requires cross-cluster access is allocated an IP, named `GlobalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This `GlobalIp` is used for all cross-cluster communication to and from to this Pod/Service. Routing and IPTable/OVS/OVN rules are configured to use this IP for ingress and egress. All address translations occur on the Gateway node.
+Once configured, each Service and Pod that requires cross-cluster access is allocated an IP, named `globalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This globalIp is used for all cross-cluster communication to and from a Pod and the globalIp of a remote Service. Routing and IPTable rules are configured to use the globalIp for ingress and egress. All address translations occur on the Gateway node.
```
[the remark this answers is not in the document]
Comment (tpantelis) [reply src/content/architecture/globalnet/_index.en.md:26]: I was wondering if `globalIp` should be capitalized too 😄 My conclusion was not to as `Pod` is an entity/object while `globalIp` is an attribute/property. But II don't know - we can  capitalized `globalIp` as well - just as long as we're connsistent.

**Pull request merged**